extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/DevServicesKafkaProcessor.java

```diff
@@ -0,0 +1,296 @@
+        if (kafkaBroker != null) {
+            closeable = kafkaBroker.getCloseable();
+            runTimeConfiguration.produce(new RunTimeConfigurationDefaultBuildItem(
+                    KAFKA_BOOTSTRAP_SERVERS, kafkaBroker.getBootstrapServers()));
+            bootstrapServers = new DevServicesKafkaBrokerBuildItem(kafkaBroker.getBootstrapServers());
+        }
+
+        // Configure the watch dog
+        if (first) {
+            first = false;
+            Runnable closeTask = new Runnable() {
+                @Override
+                public void run() {
+                    if (closeable != null) {
+                        shutdownBroker();
+                    }
+                    first = true;
+                    closeable = null;
+                    cfg = null;
+                }
+            };
+            QuarkusClassLoader cl = (QuarkusClassLoader) Thread.currentThread().getContextClassLoader();
+            ((QuarkusClassLoader) cl.parent()).addCloseTask(closeTask);
+            Thread closeHookThread = new Thread(closeTask, "Kafka container shutdown thread");
+            Runtime.getRuntime().addShutdownHook(closeHookThread);
+            ((QuarkusClassLoader) cl.parent()).addCloseTask(new Runnable() {
+                @Override
+                public void run() {
+                    Runtime.getRuntime().removeShutdownHook(closeHookThread);
+                }
+            });
+        }
+        cfg = configuration;
+
+        if (bootstrapServers != null) {
+            log.infof(
+                    "Dev Services for Kafka started. Start applications that need to use the same Kafka broker "
+                            + "using -Dkafka.bootstrap.servers=%s",
+                    bootstrapServers.getBootstrapServers());
+
+            devServicePropertiesProducer.produce(new DevServicesNativeConfigResultBuildItem("kafka.bootstrap.servers",
+                    bootstrapServers.getBootstrapServers()));
+        }
+
+        return bootstrapServers;
+    }
+
+    private void shutdownBroker() {
+        if (closeable != null) {
+            try {
+                closeable.close();
+            } catch (Throwable e) {
+                log.error("Failed to stop the Kafka broker", e);
+            } finally {
+                closeable = null;
+            }
+        }
+    }
+
+    private KafkaBroker startKafka(KafkaDevServiceCfg config) {
+        if (!config.devServicesEnabled) {
+            // explicitly disabled
+            log.debug("Not starting dev services for Kafka, as it has been disabled in the config.");
+            return null;
+        }
+
+        if (!isDockerWorking.getAsBoolean()) {
+            log.warn("Docker isn't working, please configure the Kafka bootstrap servers property (kafka.bootstrap.servers).");
+            return null;
+        }
+
+        // Check if kafka.bootstrap.servers is set
+        if (ConfigUtils.isPropertyPresent(KAFKA_BOOTSTRAP_SERVERS)) {
```

Comment:
Shouldn't this check be above the docker one? Otherwise if you have configured `kafka.bootstrap.servers` but don't have docker you will get a weird warning telling you to configure `kafka.bootstrap.servers` even though you already have.